SECURITY.md

```diff
@@ -76,4 +105,31 @@ Extensions configuration setup is tied to OpenSearch settings, extensions config
 
 ## Reliability
 
-OpenSearch plugins can create cluster or node instability if incorrectly configured or by software defects.
\ No newline at end of file
+OpenSearch plugins can create cluster or node instability if incorrectly configured or by software defects.
+
+# Projects
+To stretch out the design process while fulfilling scenarios some security efforts will be tracked as longer running projects.  There will be tracking issues in github tied to the work, documentation here is justification for a project and how it ties into the security space.  Tasks will be denotated as incomplete with :negative_squared_cross_mark: `:negative_squared_cross_mark:` or completed with :white_check_mark:
+ `:white_check_mark:`.
+
+## Anomaly Detection Plugin
+Overall project is tracked with [[FEATURE] Migrate Anomaly Detector plugin to work as an Extension](https://github.com/opensearch-project/opensearch-sdk/issues/24).  By migrating this plugin it will exercise the general extensions and security specific scenarios.
+
+Additional background avaliable from [Security#1895](https://github.com/opensearch-project/security/issues/1895)
+
+### User identity [OpenSearch#3846](https://github.com/opensearch-project/OpenSearch/issues/3846) :negative_squared_cross_mark:
```

Comment:
AD exposes a bunch of actions. 
For example: `cluster:admin/opensearch/ad/detector/delete`[1] is a Transport Action API exposed by AD. 
How do we plan to support authorization of user for extensions?

[1] https://github.com/opensearch-project/anomaly-detection/blob/4baf75ca46da02c4e70016d34982f2c5a34d0acd/src/main/java/org/opensearch/ad/transport/DeleteAnomalyDetectorAction.java#L20

Reply:
With the existing security plugin in place, no additional action is needed as these are already filtered.  The assumption is that the security plugins hooks will still execute before the extensions hooks are triggered so the API calls can be intercepted as is already implemented.

If/When we rewrite the permissions evaluation system I think part of the backwards compatibility story will need to account for scenarios like this where no/minimal action is needed by plugin authors.